docs/operations/vendors-custom-cps.md

```diff
@@ -0,0 +1,234 @@
+### Type
+Supported input field types include:
+
+- `text`: Simple plain text input.
+- `password`: Hidden secret data plain text input.
+- `select`: List of values that the user can select.
+- `select-multiple`: Allow to select multiple values from a list.
+- `json`: Object data in JSON format.
+- `yaml`: Object data in YAML input.
+
+### Values
+For the `select` and `select-multiple` input field, the list of values can be:
+
+- `Array`: A simple list of values.
+- `Object`: Use `{ cloudprofilePath, key }` - Source path in the cloud profile. If an object, use `key` to map values.
+
+
+### Validators
+| type | properties | description |
+| ---- | ---------- | ----------- |
+| `required` | `none` | Required input field |
+| `requiredIf` | `not`: Optional array of field keys. Makes field required if one of the referenced input fields is empty. | Depends on other input field if required or not. Use `not` references to create *either or* scenarios, e.g., `username / password` *or* `token` is required |
+| `regex` | `value`: JavaScript regex pattern | Input field must match the provided pattern. You can define arbitrary validations, e.g., length or specific syntax |
+| `isValidObject` | `none` | Input value must be a valid object. Can be used for `json` and `yaml` field types to ensure valid data |
+
+You can add an optional `message` property to all validators to provide a custom error message.
+
+## Example
+A complete custom cloud provider configuration example including required available cloud provider configuration as well as vendor configuration:
+
+```yaml
+global:
+  dashboard:
+    frontendConfig:
+      cloudProviderList:
+      - aws
+      - my-cc
+      - gcp
+      customCloudProviders:
+        my-cc:
+          zoned: true
+          shoot:
+            createFields:
+            - key: extra
+              path: spec.provider.extra
+              hint: Enter Additional data
+              label: Additional Information
+              type: text
+              validators:
+                required:
+                  type: required
+                isNamespace:
+                  type: regex
+                  value: .+.+--.+
+                  message: Must be valid extra data with format a--b
+            - key: cp_select
+              path: spec.provider
+              hint: Select a value from the cloud profile
+              label: Select Value
+              type: select
+              validators:
+                required:
+                  type: required
+              values:
+                cloudprofilePath: data.foo.bar
+                key: name
+            - key: value_select
+              path: spec.provider
+              hint: Select at least one predefined value
+              label: Select Value
+              type: select-multiple
+              validators:
+                required:
+                  type: required
+              values:
+                - foo
+                - bar
+            specTemplate:
```

Comment:
I could not get the `specTemplate` to work, the following was my config (snipped)

```
  customCloudProviders:
    custom-provider:
      shoot:
        specTemplate:
          provider:
            type: DUMMY
          networking:
            nodes: 9.8.7.6
```